internal/reconciliations/istio/configuration.go

```diff
@@ -46,14 +47,14 @@ func EvaluateIstioCRChanges(istioCR operatorv1alpha1.Istio, istioTag string) (tr
 	trigger = NoChange
 
 	var lastAppliedConfig appliedConfig
-	err = json.Unmarshal([]byte(lastAppliedConfigAnnotation), &lastAppliedConfig)
-	if err != nil {
-		return trigger, err
+	gErr := json.Unmarshal([]byte(lastAppliedConfigAnnotation), &lastAppliedConfig)
+	if gErr != nil {
+		return trigger, described_errors.NewDescribedError(gErr, "Could not unmarshal last applied configuration")
```

Comment:
is ti possible to cover for all these error cases in EvaluateIstioCRChanges in the unit test?